tools/nodejs/package.json

```diff
@@ -8,7 +8,7 @@
   "dependencies": {
     "@mapbox/node-pre-gyp": "^1.0.0",
     "node-addon-api": "*",
-    "node-gyp": "^7.1.2"
+    "node-gyp": "^9.3.0"
```

Comment:
The main fix in this pr - includes fixes for a number of MacOS issues, as well as fixing the 'rU' error we're seeing in the master CI